src/xrc/src/api.rs

```diff
@@ -356,19 +357,31 @@ async fn handle_cryptocurrency_pair(
     }
 
     if !utils::is_caller_privileged(&caller) {
+        let current_timestamp = env.time_secs();
+        let timestamp_is_in_future = timestamp.value > current_timestamp;
```

Comment:
I think we can be a bit more lenient: If the timestamp maps to the current minute, it is still okay:

`timestamp_is_in_future = timestamp.value / 60 > current_timestamp / 60`

Reply:
I don't think this is needed as `timestamp.value` is a normalized timestamp to the latest minute, so if it is still greater than the `current_timestamp` then `timestamp` is in the future.

Reply:
You are right. If the timestamp is normalized, the seconds are removed. If it is still larger than current_timestamp, it must be the next minute or later. So, dividing by 60 is not necessary.